x/ibcratelimit/module/ibc_middleware.go

```diff
@@ -0,0 +1,253 @@
+package module
+
+import (
+	"encoding/json"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+
+	"github.com/provenance-io/provenance/internal/ibc"
+	"github.com/provenance-io/provenance/x/ibcratelimit"
+	"github.com/provenance-io/provenance/x/ibcratelimit/keeper"
+)
+
+var (
+	_ porttypes.Middleware = &IBCMiddleware{}
+)
+
+// IBCMiddleware is the middleware used by the module.
+type IBCMiddleware struct {
+	app     porttypes.IBCModule
+	keeper  *keeper.Keeper
+	channel porttypes.ICS4Wrapper
+}
+
+// NewIBCMiddleware Creates a new IBCMiddleware.
+func NewIBCMiddleware(app porttypes.IBCModule,
+	channel porttypes.ICS4Wrapper,
+	keeper *keeper.Keeper) IBCMiddleware {
+	return IBCMiddleware{
+		app:     app,
+		keeper:  keeper,
+		channel: channel,
+	}
+}
+
+// WithIBCModule Creates a copy of the Middleware with the base application set.
+func (im *IBCMiddleware) WithIBCModule(app porttypes.IBCModule) *IBCMiddleware {
+	im.app = app
+	return im
+}
```

Comment:
Since the receiver is `*...`, this does not actually create a copy as the comment says. It updates the original and returns it.

Again, if this was just a copy/paste thing, ignore this comment and just click it as resolved.

Reply:
Forgot to update it after changing it to a pointer receiver.